modules/text2vec-databricks/clients/databricks.go

```diff
@@ -0,0 +1,233 @@
+		return nil, nil, v.getError(res.StatusCode, resBody)
+	}
+	rateLimit := ent.GetRateLimitsFromHeader(res.Header)
+
+	texts := make([]string, len(resBody.Data))
+	embeddings := make([][]float32, len(resBody.Data))
+	openAIerror := make([]error, len(resBody.Data))
+	for i := range resBody.Data {
+		texts[i] = resBody.Data[i].Object
+		embeddings[i] = resBody.Data[i].Embedding
+		// TODO: Check how Databricks Api reports errors in individual embeddings are reported in the response
+		// if resBody.Data[i].Error != nil {
+		// 	openAIerror[i] = v.getError(res.StatusCode, resBody.Data[i].Error, config.IsAzure)
+		// }
+	}
+
+	return &modulecomponents.VectorizationResult{
+		Text:       texts,
+		Dimensions: len(resBody.Data[0].Embedding),
+		Vector:     embeddings,
+		Errors:     openAIerror,
+	}, rateLimit, nil
+}
+
+func (v *client) buildURL(ctx context.Context, config ent.VectorizationConfig) (string, error) {
+	servingUrl := config.ServingURL
+	if headerServingUrl := modulecomponents.GetValueFromContext(ctx, "X-Databricks-Servingurl"); headerServingUrl != "" {
+		servingUrl = headerServingUrl
+	}
+	return servingUrl, nil
+}
+
+func (v *client) getError(statusCode int, resBody embedding) error {
+	endpoint := "Databricks Foundation Model API"
+
+	if resBody.ErrorCode != "" {
+		return fmt.Errorf("connection to: %s failed with error code: %s message: %v", endpoint, resBody.ErrorCode, resBody.Message)
+	}
+
+	if resBody.Error.Message != "" {
+		return fmt.Errorf("connection to: %s failed with status: %d error: %v", endpoint, statusCode, resBody.Error.Message)
+	}
+	return fmt.Errorf("connection to: %s failed with status: %d", endpoint, statusCode)
+}
+
+func (v *client) getEmbeddingsRequest(input []string, instruction string) embeddingsRequest {
+	return embeddingsRequest{Input: input, Instruction: instruction}
+}
+
+func (v *client) getApiKeyHeaderAndValue(apiKey string, isAzure bool) (string, string) {
+	if isAzure {
+		return "api-key", apiKey
+	}
+	return "Authorization", fmt.Sprintf("Bearer %s", apiKey)
+}
+
+func (v *client) GetApiKeyHash(ctx context.Context, cfg moduletools.ClassConfig) [32]byte {
+	key, err := v.getApiKey(ctx)
+	if err != nil {
+		return [32]byte{}
+	}
+	return sha256.Sum256([]byte(key))
+}
+
+func (v *client) GetVectorizerRateLimit(ctx context.Context, cfg moduletools.ClassConfig) *modulecomponents.RateLimits {
+	config := v.getVectorizationConfig(cfg)
+	name := "Openai"
+	if config.IsAzure {
+		name = "Azure"
+	}
+	rpm, tpm := modulecomponents.GetRateLimitFromContext(ctx, name, 0, 0)
+	return &modulecomponents.RateLimits{
+		RemainingTokens:   tpm,
+		LimitTokens:       tpm,
+		ResetTokens:       time.Now().Add(61 * time.Second),
+		RemainingRequests: rpm,
+		LimitRequests:     rpm,
+		ResetRequests:     time.Now().Add(61 * time.Second),
+	}
+}
+
+func (v *client) getApiKey(ctx context.Context) (string, error) {
+	var apiKey, envVarValue, envVar string
+
+	apiKey = "X-Databricks-Token"
```

Comment:
we should change this name to `X-Databricks-Api-Key` and should be added also [here](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L298) to the cors whitelist